accommodation/views.py

```diff
@@ -0,0 +1,95 @@
+
+class AccommodationListView(View):
+    def get(self, request):         
+        try:
+            checkin   = request.GET.get('checkin')
+            checkout  = request.GET.get('checkout')
+            guests    = request.GET.get('guests')
+            entire    = request.GET.get('entire') 
+            private   = request.GET.get('private') 
+            hotel     = request.GET.get('hotel')
+            shared    = request.GET.get('shared')
+            price_min = request.GET.get('min')
+            price_max = request.GET.get('max')
+
+            calendar_condition = Q()
+            if (checkin and checkout):
+                calendar_condition.add((
+                    Q(unavailabledate__start_date__gt= datetime.strptime(checkout, "%Y-%m-%d").date()) &
+                    Q(unavailabledate__end_date__lt=datetime.strptime(checkin, "%Y-%m-%d").date())
+                ), Q.OR)
+
+            q = Q()
+            guests_condition = Q()
+            if guests:
+                guests_condition = (Q(max_capacity__gt=guests))
+                q.add(guests_condition, Q.AND)
+
+            room_condition = Q()
+
+            if entire == 'true':
+                entire_condition  = Q(category__name='집 전체')
+                room_condition.add(entire_condition, Q.OR)
+
+            if private == 'true':
+                private_condition = Q(category__name='개인실')
+                room_condition.add(private_condition, Q.OR)
+
+            if shared == 'true':
+                shared_condition  = Q(category__name='다인실')
+                room_condition.add(shared_condition, Q.OR)
+
+            if hotel == 'true':
+                hotel_condition   = Q(category__name='호텔 객실')
+                room_condition.add(hotel_condition, Q.OR)
+                
+            q.add(room_condition, Q.AND)
+            
+            price_condition = Q()
+            if (price_min and price_max):
+                price_condition = (Q(price__lt=price_max) & Q(price__gt=price_min))
+                q.add(price_condition, Q.AND)
+            
+            accommodations = Accommodation.objects.prefetch_related('category').exclude(calendar_condition).filter(q)
+
+            data = [
+                {
+                'id'       : accommodation.id,
+                'img'      : [image.image_url for image in accommodation.image_set.all()],
+                'location' : '강남구 ' + accommodation.category.name,
+                'title'    : accommodation.title,
+                'MaxNum'   : accommodation.max_capacity,
+                'grade'    : str(round(mean(accommodation.review_set.all().aggregate(
+                    Avg('clean_rate'),
+                    Avg('communication_rate'),
+                    Avg('checkin_rate'),
+                    Avg('accuracy_rate'),
+                    Avg('location_rate'),
+                    Avg('value_rate')).values()), 2)) 
+                    if accommodation.review_set.all().exists() else '0',
```

Comment:
```suggestion
                'grade'    : str(round(mean(accommodation.review_set.all().aggregate(
                    Avg('clean_rate'),
                    Avg('communication_rate'),
                    Avg('checkin_rate'),
                    Avg('accuracy_rate'),
                    Avg('location_rate'),
                    Avg('value_rate')).values()), 2)) 
                    if accommodation.review_set.exists() else '0',
```
1. grade를 string으로 다시 형변환 하시는 이유가 있나요?
2. 지역을 강남구만 하는 것을 제가 알고 있긴한뎅, '강남구'로 하드코딩해두시는 것보다는 다른 방법을 활용해서 변수로 바꿀 수 있도록 해주세요 :)
3. `exists()` 는 존재 여부만 확인하면 되기 때문에 `all()`메소드를 사용하지 않으셔도 됩니다 :) 제가 기억하기론 `all()` 했을 때랑 `all().exists()`했을 때랑 날아가는 쿼리문이 다른 것으로 알고 있는데 확실하지는 않네요. 한 번 확인해주세요 :)

Reply:
1. 프론트에서 string으로 받고 싶다고 말씀을 주셔서 string으로 보내드리기로 했습니다. 
2. 앗 넵 시도해보겠습니다.
3. 쿼리문 찍어봤는데 all()이 있을때와 없을때의 쿼리문이 동일합니다!

Reply:
```
>>> **all() 없을때**
>>> Accommodation.objects.get(id=1).review_set.exists()
(0.011) SELECT `accommodations`.`id`, `accommodations`.`category_id`, `accommodations`.`user_id`, `accommodations`.`title`, `accommodations`.`address`, `accommodations`.`latitude`, `accommodations`.`longitude`, `accommodations`.`description`, `accommodations`.`max_capacity`, `accommodations`.`price`, `accommodations`.`cleaning_fee`, `accommodations`.`number_of_bed`, `accommodations`.`number_of_bedroom`, `accommodations`.`number_of_bathroom` FROM `accommodations` WHERE `accommodations`.`id` = 1 LIMIT 21; args=(1,)
(0.007) SELECT (1) AS `a` FROM `reviews` WHERE `reviews`.`accommodation_id` = 1 LIMIT 1; args=(1,)
True
>>>

>>> **all() 있을때**
>>> Accommodation.objects.get(id=1).review_set.all().exists()
(0.019) SELECT `accommodations`.`id`, `accommodations`.`category_id`, `accommodations`.`user_id`, `accommodations`.`title`, `accommodations`.`address`, `accommodations`.`latitude`, `accommodations`.`longitude`, `accommodations`.`description`, `accommodations`.`max_capacity`, `accommodations`.`price`, `accommodations`.`cleaning_fee`, `accommodations`.`number_of_bed`, `accommodations`.`number_of_bedroom`, `accommodations`.`number_of_bathroom` FROM `accommodations` WHERE `accommodations`.`id` = 1 LIMIT 21; args=(1,)
(0.008) SELECT (1) AS `a` FROM `reviews` WHERE `reviews`.`accommodation_id` = 1 LIMIT 1; args=(1,)
True
```